ex8/js/ajax.js

```diff
@@ -0,0 +1,18 @@
+function ajax_get(url, callback) {
+  xmlhttp = new XMLHttpRequest();
+  xmlhttp.onreadystatechange = function() {
+    if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
```

Comment:
What would happen is `readyState!==4` or `status!==200`. Right now, nothing (your callback isn't called), which is probably not good.
You can simulate this by calling `ajax_get` in the console, or by a click of a button, after you have killed your local HTTP server.